(node:483109) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'receipt' of undefined
    at exports.MiningReceipts (/home/surpgjre/SHF/app/controllers/user.controller.js:184:42)
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:137:13)
    at /home/surpgjre/SHF/app/utils/auth.js:3:39
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
    at /home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/index.js:335:12)
(node:483109) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:483109) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Error: ER_CANT_AGGREGATE_2COLLATIONS: Illegal mix of collations (latin1_swedish_ci,IMPLICIT) and (utf8_general_ci,COERCIBLE) for operation '='
    at Query.Sequence._packetToError (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
    at Query.ErrorPacket (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
    at Protocol._parsePacket (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/Parser.js:43:10)
    at Protocol.write (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/Protocol.js:38:16)
    at Socket.<anonymous> (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/Connection.js:88:28)
    at Socket.<anonymous> (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/Connection.js:526:10)
    at Socket.emit (events.js:198:13)
    at addChunk (_stream_readable.js:288:12)
    --------------------
    at Pool.query (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/Pool.js:199:23)
    at Object.exports.selectByOne (/home/surpgjre/SHF/app/models/models.js:8:7)
    at Strategy._verify (/home/surpgjre/SHF/app/utils/passport.js:13:18)
    at Strategy.authenticate (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/passport-local/lib/strategy.js:90:12)
    at attempt (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/passport/lib/middleware/authenticate.js:366:16)
    at authenticate (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/passport/lib/middleware/authenticate.js:367:7)
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
Error: ER_CANT_AGGREGATE_2COLLATIONS: Illegal mix of collations (latin1_swedish_ci,IMPLICIT) and (utf8_general_ci,COERCIBLE) for operation '='
    at Query.Sequence._packetToError (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
    at Query.ErrorPacket (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
    at Protocol._parsePacket (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/Parser.js:43:10)
    at Protocol.write (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/Protocol.js:38:16)
    at Socket.<anonymous> (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/Connection.js:88:28)
    at Socket.<anonymous> (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/Connection.js:526:10)
    at Socket.emit (events.js:198:13)
    at addChunk (_stream_readable.js:288:12)
    --------------------
    at Pool.query (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/Pool.js:199:23)
    at Object.exports.selectByOne (/home/surpgjre/SHF/app/models/models.js:8:7)
    at Strategy._verify (/home/surpgjre/SHF/app/utils/passport.js:13:18)
    at Strategy.authenticate (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/passport-local/lib/strategy.js:90:12)
    at attempt (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/passport/lib/middleware/authenticate.js:366:16)
    at authenticate (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/passport/lib/middleware/authenticate.js:367:7)
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
Error: ER_CANT_AGGREGATE_2COLLATIONS: Illegal mix of collations (latin1_swedish_ci,IMPLICIT) and (utf8_general_ci,COERCIBLE) for operation '='
    at Query.Sequence._packetToError (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
    at Query.ErrorPacket (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
    at Protocol._parsePacket (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/Parser.js:43:10)
    at Protocol.write (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/protocol/Protocol.js:38:16)
    at Socket.<anonymous> (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/Connection.js:88:28)
    at Socket.<anonymous> (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/Connection.js:526:10)
    at Socket.emit (events.js:198:13)
    at addChunk (_stream_readable.js:288:12)
    --------------------
    at Pool.query (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/mysql/lib/Pool.js:199:23)
    at Object.exports.selectByOne (/home/surpgjre/SHF/app/models/models.js:8:7)
    at Strategy._verify (/home/surpgjre/SHF/app/utils/passport.js:13:18)
    at Strategy.authenticate (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/passport-local/lib/strategy.js:90:12)
    at attempt (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/passport/lib/middleware/authenticate.js:366:16)
    at authenticate (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/passport/lib/middleware/authenticate.js:367:7)
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
(node:2398702) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'receipt' of undefined
    at exports.MiningReceipts (/home/surpgjre/SHF/app/controllers/user.controller.js:188:42)
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:137:13)
    at /home/surpgjre/SHF/app/utils/auth.js:3:39
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
    at /home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/index.js:335:12)
(node:2398702) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:2398702) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:2414474) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'receipt' of undefined
    at exports.MiningReceipts (/home/surpgjre/SHF/app/controllers/user.controller.js:188:42)
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:137:13)
    at /home/surpgjre/SHF/app/utils/auth.js:3:39
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/layer.js:95:5)
    at /home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/home/surpgjre/nodevenv/SHF/10/lib/node_modules/express/lib/router/index.js:335:12)
(node:2414474) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:2414474) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
